### DISCUSSION TITLE
Rename FPS into something more meaningful

### DISCUSSION THREAD

**Discussion by ihhub** [ihhub/fheroes2#6481] [General]
As everyone noticed many players misinterpret FPS with lower values as slow performance of the application which in most of cases just corresponds to the way the game needs to update frames. As a solution we might need to rebrand this option to avoid such thinking.

**Comment by oleg-derevenetz** [comment]
Not just in fheroes2:

https://github.com/freeorion/freeorion/issues/1291

:) IMHO the term "FPS" correctly reflects the reality, because it is literally the number of redraws per second, and these people just need some education :)

**Comment by zenseii** [comment]
Yes, but who will be appointed FPS-teacher (involuntarily?😝).
I believe I proposed this earlier, but how about a percentage? If we know the max for a case is 120 fps and you're getting 120 then that's 100%, same for 20 fps screens.

**Reply by oleg-derevenetz** [reply]
We don't know a max really. For example, on the adventure map, when player do nothing, ~20 FPS is OK, because it's the natural rate of map object animations. However, when player begins to scroll/drag the map, FPS will spike, and that's also fine. The same is with the software cursor - when player starts to move it, FPS will spike, but with the hardware cursor it will not, because it is not drawn by the game engine, but by the OS. FPS value means nothing without a context, and is intended primarily for debugging purposes. fheroes2 is not some kind of first-person shooter where "more FPS is better".

**Reply by zenseii** [reply]
So what we're interested in are skipped/lost frames, in other words that didn't get displayed? Not sure how you could get that information.

**Reply by oleg-derevenetz** [reply]
We are interested in anomalies. For example, high FPS in some static dialog is not normal (even if it doesn't affect the gameplay from the point of view of an ordinary user). There can be no such thing as "lost frames" currently, because frames are drawn immediately during the `render()` call, there is no any "frame queue" where new frames may not fit, for example, or anything like that.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub, @oleg-derevenetz and @zenseii,

I think this is more useful for "game engine behavior" testing than anything else.
I don't know if this is feasible, but why not display two values?
The actual value (first) and the desired/estimated/optimal value (second):
FPS: 100/98, FPS: 12/14
or:
FPS: 100 (98), FPS: 12 (14)

**Reply by zenseii** [reply]
Hi, @LeHerosInconnu, @oleg-derevenetz pointed out above why it can't be done this way.

**Comment by ihhub** [comment]
As @oleg-derevenetz , FPS measure is more for development purposes rather than a measure how stable the game is. It was introduced to see how performance improvements affect the rendering. Many players misunderstand this value and think that 20 FPS on Adventure Map is when the game is underperforming which is not true. Based on this conversation I think we at least need to expand the description of the option to state that the game has variable FPS.
